### ISSUE TITLE
Protractor 2.0.0 with directConnect doesn't work on Firefox 38

### ISSUE THREAD

**Issue by mzgol** [angular/protractor#2134]
Firefox 38 is out; trying to run Protractor `2.0.0` on it results in the following output:
```
[firefox #2] PID: 81496
[firefox #2] Using FirefoxDriver directly...
[firefox #2] ERROR - Unable to start a WebDriver session.
[firefox #2] 
[firefox #2] /Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877
[firefox #2]       var result = fn();
[firefox #2]                    ^
[firefox #2] Error: Failed to install profile; firefox terminated with Result(code=null, signal=SIGSEGV)
[firefox #2]     at Error (native)
[firefox #2]     at /Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/firefox/binary.js:239:67
[firefox #2]     at [object Object].promise.ControlFlow.runInFrame_ (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877:20)
[firefox #2]     at [object Object].promise.Callback_.goog.defineClass.notify (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:2464:25)
[firefox #2]     at [object Object].promise.Promise.notify_ (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:563:12)
[firefox #2]     at Array.forEach (native)
[firefox #2]     at Object.goog.array.forEach (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/array/array.js:203:43)
[firefox #2]     at [object Object].promise.Promise.notifyAll_ (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:552:16)
[firefox #2]     at goog.async.run.processWorkQueue (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/async/run.js:125:21)
[firefox #2]     at process._tickCallback (node.js:364:9)
[firefox #2] From: Task: WebDriver.createSession()
[firefox #2]     at Function.webdriver.WebDriver.acquireSession_ (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:155:22)
[firefox #2]     at Function.webdriver.WebDriver.createSession (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:129:30)
[firefox #2]     at new Driver (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/selenium-webdriver/firefox/index.js:272:36)
[firefox #2]     at [object Object].DirectDriverProvider.getNewDriver (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/lib/driverProviders/direct.js:75:16)
[firefox #2]     at [object Object].Runner.createBrowser (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/lib/runner.js:180:37)
[firefox #2]     at /Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/lib/runner.js:257:21
[firefox #2]     at _fulfilled (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/q/q.js:797:54)
[firefox #2]     at self.promiseDispatch.done (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/q/q.js:826:30)
[firefox #2]     at Promise.promise.promiseDispatch (/Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/q/q.js:759:13)
[firefox #2]     at /Users/mgol/Documents/projects/bn/cbn/repo/polona-gui/node_modules/protractor/node_modules/q/q.js:573:44

[launcher] Runner process exited unexpectedly with error code: 1
```

**Comment by Tekki** [comment]
Can confirm this for Ubuntu 12.04. On Windows 8.1 everything works fine.
There was a similar issue on Selenium: https://github.com/SeleniumHQ/selenium/issues/437
But the solution there doesn't work for me. Firefox 38 doesn't start even if I installed it in a separate directory and set the firefoxPath parameter. All versions below 38 start normally.

**Comment by plynchnlm** [comment]
A coworker found that downgrading to protractor 1.8 was a workaround.

**Comment by jodytate** [comment]
I'm seeing the same issue without Selenium while using the `directConnect: true` option.

**Comment by mzgol** [comment]
Indeed, this is because of `directConnect: true`! I've disabled it now and
it works fine for me.

**Comment by NickTomlin** [comment]
I'm also seeing this with `directConnect: true` with FF 38.0.1 on OSX 0.10.3.

**Comment by Nopik** [comment]
Actually, I'm seeing the same problem with protractor 1.8.0, if I have `directConnect: true`, it fails, with `false` it works.

**Comment by mchapman** [comment]
It looks to me like this could now be addressed, since https://github.com/SeleniumHQ/selenium/issues/437 (which is presumably the cause) was fixed back in May.

**Comment by hankduan** [comment]
This is a known issue (https://github.com/SeleniumHQ/selenium/issues/437). It is fixed by selenium and tagged for released at 2.47.1. https://github.com/SeleniumHQ/selenium/commit/144f5ee7eb7d705bad90346bb2b494ff8212fb4c

**Comment by NickTomlin** [comment]
@hankduan awesome. Thanks for the update!

**Comment by juliemr** [comment]
This is fixed with 2.4.0
